commands/init.go

```diff
@@ -188,18 +188,58 @@ func loadGenesis(ctx context.Context, rep repo.Repo, sourceName string) (consens
 		return nil, err
 	}
 
-	if len(ch.Roots) != 1 {
-		return nil, fmt.Errorf("expected car with only a single root")
+	// need to check if we are being handed a car file with a single genesis block or an entire chain.
```

Comment:
Do we want to support initializing with a CAR file that has all state tree data as well?

Reply:
I don't think any special handling should be required (at least not here). As I understand it, `car.LoadCar` will load all blocks from the CAR into the store: blockchain, messages, receipts, state trees, whatever it has.

Reply:
Sorry I should have been more clear here, there are 4 cases I can think of for how a node could be initialized:
1. No genesis block is passed, just a vanilla `go-filecoin init`
2. A genesis block/CAR-file is passed:
  a. It contains only genesis block and state.
  b. It contains a chain, genesis block, and its state.
  c. It contains a chain with all state for every transition. 

go-filecoin has cases 1 and 2.a covered, and the code added here mostly addresses 2.b- all that's left is to sync up to the head returned by the modified genesis function here. 
I believe 2.c may require some special handing here to communicate to the chain init method that it should persist the state roots contained in the CAR file to its chain store.

I am realizing now that adding support for 2.c (if we choose to do so) is outside the scope of this PR and have filed https://github.com/filecoin-project/go-filecoin/issues/3606.